src/plenigo/services/MobileService.php

```diff
@@ -0,0 +1,186 @@
+ */
+class MobileService extends Service {
+
+    const ERR_MSG_VERIFY = "Error during mobile secret verification";
+    const ERR_MSG_GET = "Error geting mobile secret";
+    const ERR_MSG_POST = "Error creating mobile secret";
+    const ERR_MSG_DELETE = "Error deleting mobile secret";
+
+    /**
+     * The constructor for the MobileService instance.
+     *
+     * @param RestClient $request   The RestClient request to execute.
+     *
+     * @return MobileService instance.
+     */
+    public function __construct($request) {
+        parent::__construct($request);
+    }
+
+    /**
+     * This method allows to verify if mobile app has access to certain customer
+     * 
+     * @param string $email The user email
+     * @param string $mobileSecret The mobile secret to verify
+     * @return string The Customer ID for that mobile secret
+     * @throws PlenigoException
+     */
+    public static function verifyMobileSecret($email, $mobileSecret) {
+        $map = array(
+            'companyId' => PlenigoManager::get()->getCompanyId(),
+            'secret' => PlenigoManager::get()->getSecret(),
+            'email' => $email,
+            'mobileSecret' => $mobileSecret
+        );
+
+        $url = ApiURLs::MOBILE_SECRET_VERIFY;
+
+        $request = static::getRequest($url, false, $map);
+
+        $appTokenRequest = new static($request);
+
+        $data = parent::executeRequest($appTokenRequest, ApiURLs::MOBILE_SECRET_VERIFY, self::ERR_MSG_VERIFY);
+
+        if (isset($data->customerId)) {
+            $result = $data->customerId;
+        } else {
+            $result = "" . $data;
+        }
+
+        return $result;
+    }
+
+    /**
+     * Get the mobile secret for a given Customer ID
+     * 
+     * @param string $customerId The Customer ID
+     * @return MobileSecretData the email and secret for the customer mobile
+     * @throws PlenigoException
+     */
+    public static function getMobileSecret($customerId) {
+        $map = array(
+            'companyId' => PlenigoManager::get()->getCompanyId(),
+            'secret' => PlenigoManager::get()->getSecret()
+        );
+
+        $url = str_ireplace(ApiParams::URL_USER_ID_TAG, $customerId, ApiURLs::MOBILE_SECRET_URL);
+
+        $request = static::getRequest($url, false, $map);
+
+        $appTokenRequest = new static($request);
+
+        $data = parent::executeRequest($appTokenRequest, ApiURLs::MOBILE_SECRET_URL, self::ERR_MSG_GET);
+
+        $result = MobileSecretData::createFromMap((array) $data);
+
+        return $result;
+    }
+
+    /**
+     * Creates a mobile secret for a given Customer ID
+     * 
+     * @param string $customerId The Customer ID
+     * @param int $size The size of the mobile secret 8 to 40
+     * 
+     * @return MobileSecretData the email and secret for the customer mobile
+     * 
+     * @throws PlenigoException
+     */
+    public static function createMobileSecret($customerId, $size) {
+        if ($size < 8) {
```

Comment:
6 would be good as lower value. And also a constant for that would be useful ;-)